snap/snapcraft.yaml

```diff
@@ -17,17 +17,27 @@ apps:
     rocketchat-server:
         command: startRocketChat
         daemon: simple
+        after: [rocketchat-mongo-daemon]
         plugs: [network, network-bind, removable-media]
+    rocketchat-mongo-daemon:
+        command: mongo.sh --start
+        stop-command: mongo.sh --stop
+        daemon: forking
+        plugs:
+            - network
+            - network-bind
+            - network-observer
     rocketchat-mongo:
-        command: startmongo
+        command: mongo.sh --logs
+        stop-command: mongo.sh --stop
+        after: [rocketchat-mongo-daemon]
         daemon: simple
-        plugs: [network, network-bind, network-observe]
```

Comment:
can you explain more why to have 2 services here?  Why fork?  Why not let systemd fully manage mongo?  Having a whole separate daemon just to run tail seems pretty wasteful.  Logs showing up in two locations

Reply:
Logs won't show up in two locations. 

By using fork I'm making sure that no other following unit (rocketchat-server) starts up before mongo is ready to accept connections. This is because the parent mongo process exits **after** the child process (or the main mongod process) is ready, and the unit is not considered to have completed startup until the parent process has exited (thanks to the service type being "forking").

Main logs are not going to show up in rocketchat-mongo-daemon, the only thing it's going to show is the PID of the child process - 

```
about to fork child process, waiting until server is ready for connections.
forked process: 15
child process started successfully, parent exiting
```

This is why the additional tailing. Since we're using systemd services, it made sense to have the logs accessible through journalctl and snap logs.

Reply:
right.. 🤔 I guess with forking we can't use a single systemd service and have logs spit out?  Having to have a dedicated service just for logs seems like a design we don't want.  Will make things very confusing for our users

Reply:
I was afraid of the same. Had something else in mind but that didn't work as expected :(